File: solutions/javascript/CountPrimes/CountPrimes.js
/**
 * @param {number} n
 * @return {number}
 */
var countPrimes = function(n) {
    let count = 0, arr = Array(n).fill(false);
    for (let i = 2; i < n; i++) {
        if (!arr[i]) {
            count++;
            for (let j = 2; i * j < n; j++) {
                arr[i * j] = true;
            }
        }
    }
    return count;
};

File: solutions/java/CountPrimes/CountPrimes.java
class Solution {
    public int countPrimes(int n) {
        int count = 0;
        int[] arr = new int[n];
        for (int i = 2; i < n; i++) {
            if (arr[i] == 0) {
                count++;
                for (int j = 2; i * j < n; j++) {
                    arr[i * j] = 1;
                }
            }
        }
        return count;
    }
}